### ISSUE TITLE
Component Data json format

### ISSUE THREAD

**Issue by michaeltintiuc** [vuejs/vetur#2179]
- [x] I have searched through existing issues
- [x] I have read through [docs](https://vuejs.github.io/vetur)
- [x] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)

I've read the Component Data docs and have looked through 3rd party implementations of attributes.json and tags.json, but I couldn't find a way to allow for auto-completion and suggestion of custom events, I've tried prefixing props with `v-on:` and `@` but it's not quite the same when you type `@` and it brings up a list of all available events to listen on.

**Comment by rchl** [comment]
I think the current format is very limited. See https://github.com/vuejs/vetur/issues/2090 for something more complete.

**Comment by octref** [comment]
I'll close after adding some doc.

**Comment by michaeltintiuc** [comment]
Thanks @octref , just to clarify, an event type would allow for auto-completion and suggestions with `v-on` and `@`?

**Comment by octref** [comment]
@michaeltintiuc Yes, sorry for the lack of documentation now...Take a look at https://github.com/vuejs/vetur/pull/2180/commits/1ccb6ba88898f003f484842be66c70c7e9c37d9d
and the setup here: https://github.com/vuejs/vetur/blob/813a3b52157e7a79a17383a320eac88f85b2050f/test/componentData/fixture/package.json#L10-L13

I'll make a release this week.

**Comment by octref** [comment]
Cool modeling btw 🎩

**Comment by michaeltintiuc** [comment]
Thanks a lot @octref !

**Comment by michaeltintiuc** [comment]
Is this scheduled for an upcoming release? If there's anything I can help with I'd be happy to

**Comment by michaeltintiuc** [comment]
I was also wondering if there's any reason to support `events` prop in `tags.json` same it's done for `attributes`, i.e.:
```json
{
  "foo": {
    "description": "Foo tag",
    "attributes": ["bar"],
    "events": ["baz"]
  }
}
```
